zebra-state/src/service/check/tests/anchors.rs

```diff
@@ -11,7 +13,118 @@ use crate::{
     tests::setup::{new_state_with_mainnet_genesis, transaction_v4_from_coinbase},
 };
 
-// sapling
+// Sprout
+
+/// Check that, when primed with the first blocks that contain JoinSplits on
+/// Groth16, a Sprout Spend's referenced anchor is validated.
+#[test]
+fn check_sprout_anchors() {
+    zebra_test::init();
+
+    let (mut state, _genesis) = new_state_with_mainnet_genesis();
+
+    // Bootstrap a block at height == 1 that has the first Sprout note
+    // commitments on Groth16.
+    let mut block1 = zebra_test::vectors::BLOCK_MAINNET_1_BYTES
+        .zcash_deserialize_into::<Block>()
+        .expect("block should deserialize");
+
+    // Convert the coinbase transaction to a version that the non-finalized state will accept.
+    block1.transactions[0] = transaction_v4_from_coinbase(&block1.transactions[0]).into();
+
+    // Prime finalized state with the Sapling start + 1, which has the first
+    // JoinSplits on Groth16.
+    let block_419201 = zebra_test::vectors::BLOCK_MAINNET_419201_BYTES
+        .zcash_deserialize_into::<Block>()
+        .expect("block should deserialize");
```

Comment:
OK, idea: instead of taking the Sprout note commitments from the first block with `JoinSplit<Groth16>`, we can construct artificial `JoinSplit<Groth16>` transactions from the Sprout note commitments and derived anchors from the first N blocks on mainnet since genesis (genesis and block 396), and check that  `validate_and_commit()` and the `anchor` check underneath pass. We may to get the second block in mainnet to make sure we exercise looking up anchors that aren't just referencing the empty root, but first things first.

@upbqdn whatcha think?

Reply:
Makes sense. On it now.

Reply:
Thinking about it again, creating JoinSplits with empty Groth16 proofs will make further checks fail once we start validating those proofs.

Reply:
Proofs are verified during semantic verification in zebra-consensus, long before we test contextual verification and if passing push to state.